packages/compass-e2e-tests/tests/atlas-cloud/global-writes.test.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  DEFAULT_CONNECTION_NAMES,
+  isTestingAtlasCloudSandbox,
+} from '../../helpers/test-runner-context';
+
+type GeoShardingFormData = {
+  secondShardKey: string;
+  keyType?: 'UNIQUE' | 'HASHED';
+};
+
+type GeoShardingState = 'UNSHARDED' | 'SHARDING' | 'SHARD_KEY_CORRECT';
+
+async function createGeoShardKey(
+  browser: CompassBrowser,
+  formData: GeoShardingFormData
+) {
+  // shard-collection-form
+  await browser.setComboBoxValue(
+    Selectors.GlobalWrites.ShardKeyFormSecondKeyInputCombobox,
+    formData.secondShardKey
+  );
+
+  if (formData.keyType) {
+    await browser.clickVisible(
+      Selectors.GlobalWrites.ShardKeyFormAdvancedOptionsToggle
+    );
+    await browser.clickParent(
+      Selectors.GlobalWrites.shardKeyFormIndexType(formData.keyType)
+    );
+  }
+  await browser.clickVisible(Selectors.GlobalWrites.ShardKeyFormSubmitButton);
+}
+
+async function waitForGlobalWritesStatus(
+  browser: CompassBrowser,
+  nextState: GeoShardingState
+) {
+  await browser.waitUntil(async () => {
+    const content = await browser.$(
+      Selectors.GlobalWrites.tabStatus(nextState)
+    );
+    return await content.isDisplayed();
+  });
+}
+
+describe('Global writes', function () {
+  let compass: Compass;
+  let browser: CompassBrowser;
+
+  before(async function () {
+    compass = await init(this.test?.fullTitle());
+    browser = compass.browser;
+    await browser.setupDefaultConnections();
+  });
+
+  before(function () {
+    if (!isTestingAtlasCloudSandbox()) {
+      this.skip();
+    }
+  });
+
+  after(async function () {
+    await cleanup(compass);
+  });
+
+  it('should be able to shard an unsharded namespace and also unmanage it', async function () {
+    // Sharding a collection takes a bit longer
+    this.timeout(60_000);
+
+    await createGeospatialCollection();
+    await browser.connectToDefaults();
+    await browser.navigateToCollectionTab(
+      DEFAULT_CONNECTION_NAMES[0],
+      'test',
+      'geospatial',
+      'GlobalWrites'
+    );
+
+    // Initial state is loading
+    await waitForGlobalWritesStatus(browser, 'UNSHARDED');
+
+    await createGeoShardKey(browser, {
+      secondShardKey: 'country',
+      keyType: 'HASHED',
+    });
+
+    // Wait for the shard key to be correct.
+    await waitForGlobalWritesStatus(browser, 'SHARD_KEY_CORRECT');
+
+    // Expectations to see the shard key in the UI
+    const findingDocumentsText = await browser
+      .$(Selectors.GlobalWrites.SampleFindingDocuments)
+      .getText();
+
+    const insertedDocumentsText = await browser
+      .$(Selectors.GlobalWrites.SampleInsertingDocuments)
+      .getText();
+
+    expect(findingDocumentsText).to.include('country');
+    expect(insertedDocumentsText).to.include('country');
+
+    // Unmanage the namespace
+    await browser.clickVisible(Selectors.GlobalWrites.UnmanageNamespaceButton);
+
+    // It transitions to the unmanaging state
+    await waitForGlobalWritesStatus(browser, 'UNSHARDED');
```

Comment:
note: this will be `INCOMPLETE_SHARDING_SETUP` after https://github.com/mongodb-js/compass/pull/6399, but it's quite likely that this will be merged first, so I'll update the test afterwards

Reply:
I might even add a test to check that the form is not present at this point, it's important that the key cannot be changed after unmanaging. But again, this hangs on my PR.